File: src/OS/runtime/statemanager.py
from OS.domain import AgentInstance
from OS.runtime.agent import ExecutionResult


class StateManager:
    """
    agent状态管理机
    """

    # 目前只负责result写回State

    def apply(self, agent: AgentInstance, result: ExecutionResult) -> None:
        agent.state.update(result.state_update)

File: go/src/runtime/statemanager.go
package runtime

import "agent-runtime-go/src/domain"

// StateManager 统一将 Agent 返回的状态增量写回 Agent。
// 当前是内存实现；以后可在这里替换为带事务的数据库提交。
type StateManager struct{}

func (StateManager) Apply(agent *domain.AgentInstance, result ExecutionResult) {
	if agent.State == nil {
		agent.State = make(map[string]any)
	}
	for key, value := range result.StateUpdate {
		agent.State[key] = value
	}
}
